src/systemTest/java/com/flightstats/hub/system/functional/storage/BatchItemStorageTest.java

```diff
@@ -0,0 +1,73 @@
+package com.flightstats.hub.system.functional.storage;
+
+import com.flightstats.hub.kubernetes.HubLifecycle;
+import com.flightstats.hub.model.Channel;
+import com.flightstats.hub.model.ChannelStorage;
+import com.flightstats.hub.system.config.DependencyInjector;
+import com.flightstats.hub.system.service.ChannelService;
+import com.flightstats.hub.system.service.S3Service;
+import lombok.extern.slf4j.Slf4j;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import static com.flightstats.hub.util.StringUtils.randomAlphaNumeric;
+import static com.flightstats.hub.system.config.PropertiesName.TEST_DATA;
+
+@Slf4j
+class BatchItemStorageTest extends DependencyInjector {
+    @Inject
```

Comment:
Might it make sense to use parameterized tests for BATCH, BOTH, SINGLE tests and have them all in one file? It might make it easier to see the differences in how we expect each scenario to behave (in finding stuff in the hub/spoke and how things are stored in S3). It'd possibly make the tests themselves more obtuse, though, so maybe it's not worth it.

Reply:
parameterized this